token-list/utils.ts

```diff
@@ -0,0 +1,136 @@
+import { default as alephiumTestnetContracts } from '../configs/alephium/testnet.json'
+import { default as alephiumMainnetContracts } from '../configs/alephium/mainnet.json'
+import { ALPH_TOKEN_ID, hexToBinUnsafe, isHexString, web3 } from '@alephium/web3'
+import { ChainId, CHAIN_ID_ALEPHIUM, CHAIN_ID_BSC, CHAIN_ID_ETH, ethers_contracts, EVMChainId, getForeignAssetAlephium, isEVMChain } from '@alephium/wormhole-sdk'
+import { default as ethTestnetContracts } from '../configs/ethereum/testnet.json'
+import { default as ethMainnetContracts } from '../configs/ethereum/mainnet.json'
+import { default as bscTestnetContracts } from '../configs/bsc/testnet.json'
+import { default as bscMainnetContracts } from '../configs/bsc/mainnet.json'
+import { ethers } from 'ethers'
+import { getForeignAssetEth } from '@alephium/wormhole-sdk/lib/cjs/nft_bridge'
+import { ALPH } from '@alephium/token-list'
+
+export interface BridgeChain {
+  chainId: ChainId
+
+  validateAndNormalizeTokenId(tokenId: string): Uint8Array
+  getForeignAsset(tokenChainId: ChainId, tokenId: Uint8Array): Promise<string | null>
+  getTokenMetadata(tokenId: string): Promise<{ name: string, symbol: string, decimals: number }>
+}
+
+export type TokenMetaData = { name: string, symbol: string, decimals: number }
+
+export function createAlephium(network: 'testnet' | 'mainnet'): BridgeChain {
+  const deployments = network === 'testnet' ? alephiumTestnetContracts : alephiumMainnetContracts
+  const nodeUrl = network === 'testnet' ? 'https://wallet-v20.testnet.alephium.org' : 'https://wallet-v20.mainnet.alephium.org'
+  const textDecoder = new TextDecoder()
+  web3.setCurrentNodeProvider(nodeUrl)
+
+  const getForeignAsset = async (tokenChainId: ChainId, tokenId: Uint8Array): Promise<string | null> => {
+    return await getForeignAssetAlephium(
+      deployments.contracts.tokenBridge,
+      web3.getCurrentNodeProvider(),
+      tokenChainId,
+      tokenId,
+      deployments.groupIndex
+    )
+  }
+
+  const getTokenMetadata = async (tokenId: string): Promise<TokenMetaData> => {
+    if (tokenId === ALPH_TOKEN_ID) {
+      return { name: ALPH.name, symbol: ALPH.symbol, decimals: ALPH.decimals }
+    }
+    const nodeProvider = web3.getCurrentNodeProvider()
```

Comment:
Do you think it makes sense to consult [Token List](https://github.com/alephium/token-list) first here?

Reply:
Good idea, I have added a check to verify if the token is in the token list, but the token metadata is still queried from the chain.